TWNIA2/2. Tutorials/4. 存取雲端物件儲存 (COS).md

```diff
@@ -1 +1,273 @@
+$ twccli cp cos -bkt mytwccbucket -fn myfile -sync to-cos
+```
+- **View file**
+
+Check if the upload is successful
+```bash
+# for S3cmd
+$ s3cmd ls s3://mytwccbucket/myfile
+
+# for TWCC CLI
+$ twccli ls cos -bkt mytwccbucket
+```
+- **Delte file**
+
+Delete the just uploaded myfile
+```bash
+# for S3cmd
+$ s3cmd rm s3://mytwccbucket/myfile
+
+# for TWCC CLI
+$ twccli rm cos -bkt mytwccbucket -okey myfile
+```
+- **Delete bucket**
+
+Please make sure the bucket is empty before deleting bucket
+
+```bash
+# for S3cmd
+$ s3cmd rb s3://mytwccbucket
+
+# for TWCC CLI
+$ twccli rm cos -bkt mytwccbucket
+```
+
+:::info
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** Other parameters and description:
+- S3cmd
+May refer using `$ s3cmd --help` command or look into [<ins>s3cmd GitHub</ins>](https://github.com/s3tools/s3cmd#simple-s3cmd-howto).
+- TWCC CLI
+May refer using `$ twccli --help` command or look into [<ins>TWCC CLI document</ins>](https://man.twcc.ai/@twccdocs/twcc-cli-v05#4-%E9%9B%B2%E7%AB%AF%E7%89%A9%E4%BB%B6%E5%84%B2%E5%AD%98%E6%9C%8D%E5%8B%99COS-Cloud-Object-Storage).
+:::
+
+## 2. Mount
+
+The following will introduce two mounting tools: [Goofys](https://github.com/kahing/goofys) and [S3fs](https://github.com/s3fs-fuse/s3fs-fuse), you may choose according to your needs.
+
+:::info
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** The mounting method only able to used on ln01.twcc.ai login node.
+:::
+
+
+### Create a COS bucket
+
+- Please refer to TWCC [Cloud Object Storage](https://www.twcc.ai/doc?page=object) manual and create a COS bucket on portal (The management of Bucket cannot operate through TWNIA2).
+- Create a `~/mount_cos` folder after login to ln01.twcc.ai:
+
+```bash
+$ mkdir -p ~/mount_cos
+```
+
+
+### Enter key
+
+Create a folder under `~/.aws` directory, and copy TWCC COS key to `~/.aws/credentials`.
+
+
+In here using vim as an example, you may use your familiar editor.
+
+```bash
+$ mkdir -p ~/.aws
+$ vim ~/.aws/credentials
+```
+
+Enter the content below:
+
+```bash
+[default]
+aws_access_key_id = COS Access Key
+aws_secret_access_key = COS Secret Key
+```
+
+
+### [Goofys](https://github.com/kahing/goofys) 
+- Written in Go language, good performance, should have a high priority.
+
+
+Mount command:
+```bash
+$ goofys --endpoint https://cos.twcc.ai mytwccbucket ~/mount_cos
+```
+
+### [S3fs](https://github.com/s3fs-fuse/s3fs-fuse)
+- S3fs is a well-known s3 mounting tool.
+
+Mount command:
+
+```bash
+$ s3fs mytwccbucket ~/mount_s3 -o url=https://cos.twcc.ai/ -o use_path_request_style
+```
+
+### Access method
+
+After the mount is completed, you may access to the new created COS `mytwccbucket` file under the `~/mount_cos` directory.
+
+Example of operation
+
+```bash
+# Write into an empty file named myfile
+$ touch ~/mount_cos/myfile
+```
+At this time, you may click on `mytwccbucket` on **Cloud Object Storage Management** webpage to view the content of the file `myfile`.
```

Comment:
At this time, you may >

Now you can

Reply:
on **Cloud Object Storage Management** webpage > 

on the **Cloud Object Storage Management** page on the TWCC portal